gef.py

```diff
@@ -689,10 +689,30 @@ def size(self) -> int:
             raise AttributeError
         return self.page_end - self.page_start
 
+    def search_for_realpath(self) -> str:
+        # This function is a workaround for gdb bug #23764
+        # path might be wrong for remote sessions, so try a simple search for files
+        # that aren't found at the path indicated, which should be canonical.
+
+        # First, try the canonical path in the remote session root.
+        candidate1 = f"{gef.session.remote.root}/{self.path}"
+        if pathlib.Path(candidate1).is_file():
```

Comment:
Slightly better to do
```python
candidate1 = pathlib.Path(gef.session.remote.root) / self.path
```
Since it doesn't require building strings and doesn't use os specific path separators.

Reply:
Can't do that, because self.path is an absolute path, which causes the initial path to be ignored.

Still, might not be idiomatic.  Could use .join() if that's preferred.

Reply:
Updated this section using pathlib more.